### PR TITLE
pytest: Fix typo and add logging

### PR DESCRIPTION
**Pull request by dreiss**

### CODE REVIEW

**Comment by dreiss** [comment]
This is meant to be rebased and merged as two separate commits.  Please let me know if I didn't format the PR properly for that.

File: scripts/pylib/twister/twisterlib/harness.py
Comment (aaronemassey) [inline scripts/pylib/twister/twisterlib/harness.py:162]: I'm not a particular owner of this file, but I'd be hesitant to add a class here for a single example that only shows string formatting.

**Review by aaronemassey — CHANGES_REQUESTED** [review]

File: scripts/pylib/twister/twisterlib/harness.py
Comment (yperess) [reply scripts/pylib/twister/twisterlib/harness.py:162]: I'm not seeing the benefit of making a class from this, can be done inline on line 218

**Review by yperess — CHANGES_REQUESTED** [review]

**Comment by dreiss** [comment]
Doing it inline forces all of the string formatting to be done even if debug logs are turned off.  Putting it in an object delays the formatting and skips it all together if the log is not rendered.  In general, I prefer not to do expensive operations in debug logs for performance reasons, but this is not a performance-critical codepath, so maybe it would be okay.  Please let me know your preference.

**Comment by aaronemassey** [comment]
> Doing it inline forces all of the string formatting to be done even if debug logs are turned off

That's a good point! Since we're not just using the string formatting provided by the logger we don't get the benefit of lazy formatting from the logger.

However:

> but this is not a performance-critical codepath

I'd rather have a nominal increase in compute cost, especially if not performance-critical, instead of adding an additional type.

If this changes we can always add the class or use [lazy-string](http://go/pypi/lazy-string) but I don't foresee this to happen.

**Comment by dreiss** [comment]
Updated.  Is this ready to rebase+merge?

**Comment by aaronemassey** [comment]
> Updated. Is this ready to rebase+merge?

Fine with me, but you should get a review from a proper maintainer of twister. I'd recommend the pr help channel in the discord.

**Review by aaronemassey — DISMISSED** [review]

**Comment by dreiss** [comment]
@nashif is listed as the Twister maintainer.

**Comment by dreiss** [comment]
@nashif or @yperess , is this ready to be merged?

**Review by yperess — DISMISSED** [review]

**Comment by dreiss** [comment]
Looks like it was fixed by rebasing.  Can this be merged?

**Review by yperess — APPROVED** [review]

**Review by aaronemassey — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**
